src/citrine/resources/file_link.py

```diff
@@ -0,0 +1,111 @@
+"""A collection of FileLink objects."""
+from uuid import UUID
+import os
+import mimetypes
+from boto3 import client as boto3_client
+from botocore.exceptions import ClientError
+
+from taurus.entity.file_link import FileLink as TaurusFileLink
+from citrine._rest.collection import Collection
+from citrine._rest.resource import Resource
+from citrine._session import Session
+
+
+class FileLink(Resource['FileLink'], TaurusFileLink):
+    """Resource that stores the name and url of an external file."""
+
+    pass
+
+
+class FileCollection(Collection[FileLink]):
+    """Represents the collection of all file links associated with a dataset."""
+
+    _path_template = 'projects/{project_id}/datasets/{dataset_id}/files'
+    _dataset_agnostic_path_template = 'projects/{project_id}/files'
```

Comment:
No such route exists in ProductApi. Do you expect it to or is this for the future?

Reply:
No. Because so many resources have two access routes, one with a dataset and one without, I hard-coded these two paths in. But this one is not needed here and can stay NotImplemented. I'll remove it.